universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockPredicatesEditView.java

```diff
@@ -197,7 +220,35 @@ private ContainerTag renderQuestionDefinitionBox(QuestionDefinition questionDefi
                         .withClasses(Styles.MT_1, Styles.TEXT_SM)));
   }
 
-  private static ContainerTag createScalarDropdown(ImmutableMap<Scalar, ScalarType> scalars) {
+  private ContainerTag createActionDropdown(String blockName) {
+    ImmutableMap<String, String> actionOptions =
+        Arrays.stream(PredicateAction.values())
+            .collect(toImmutableMap(PredicateAction::toDisplayString, PredicateAction::name));
+    return new SelectWithLabel()
+        .setFieldName("predicateAction")
+        .setLabelText(String.format("%s should be", blockName))
+        .setOptions(actionOptions)
+        .getContainer();
+  }
+
+  private Tag createHiddenQuestionDefinitionInput(QuestionDefinition questionDefinition) {
```

Comment:
suggestion: can we add a comment explaining why this is required?

Reply:
Yes... good idea.